[CI] GenerationalDocValuesIT testBackgroundMergeCanRetainDeletedGenerationalFile failing

**Build Scans:**
- [elasticsearch-intake #40053 / part2](https://gradle-enterprise.elastic.co/s/isvj7eovtm2i6)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.lucene.GenerationalDocValuesIT.testBackgroundMergeCanRetainDeletedGenerationalFile" -Dtests.seed=E4CF6015C441C590 -Dtests.locale=kam-Latn-KE -Dtests.timezone=Europe/Vatican -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.lucene.GenerationalDocValuesIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testBackgroundMergeCanRetainDeletedGenerationalFile'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: safeAwait: CountDownLatch did not reach zero within the timeout
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testBackgroundMergeCanRetainDeletedGenerationalFile
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -498,9 +498,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.StatelessRealTimeGetIT
   method: testStress
   issue: https://github.com/elastic/elasticsearch/issues/147850
-- class: org.elasticsearch.xpack.stateless.lucene.GenerationalDocValuesIT
-  method: testBackgroundMergeCanRetainDeletedGenerationalFile
-  issue: https://github.com/elastic/elasticsearch/issues/149626
 - class: org.elasticsearch.xpack.ml.integration.RevertModelSnapshotIT
   method: testRevertModelSnapshot
   issue: https://github.com/elastic/elasticsearch/issues/147383

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/lucene/GenerationalDocValuesIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/lucene/GenerationalDocValuesIT.java
@@ -57,6 +57,7 @@ import org.elasticsearch.xpack.stateless.cache.reader.CacheBlobReaderService;
 import org.elasticsearch.xpack.stateless.cache.reader.MeteringCacheBlobReader;
 import org.elasticsearch.xpack.stateless.cache.reader.MutableObjectStoreUploadTracker;
 import org.elasticsearch.xpack.stateless.cache.reader.ObjectStoreCacheBlobReader;
+import org.elasticsearch.xpack.stateless.cluster.coordination.StatelessClusterConsistencyService;
 import org.elasticsearch.xpack.stateless.commits.BatchedCompoundCommit;
 import org.elasticsearch.xpack.stateless.commits.BlobFile;
 import org.elasticsearch.xpack.stateless.commits.BlobFileRanges;
@@ -478,6 +479,7 @@ public class GenerationalDocValuesIT extends AbstractStatelessPluginIntegTestCas
                 .put(StatelessCommitService.STATELESS_UPLOAD_MAX_AMOUNT_COMMITS.getKey(), 100)
                 // Need at least 2 threads as we will block two
                 .put("thread_pool." + ThreadPool.Names.MERGE + ".max", randomIntBetween(2, 4))
+                .put(StatelessClusterConsistencyService.DELAYED_CLUSTER_CONSISTENCY_INTERVAL_SETTING.getKey(), "100ms")
                 .build()
         );
 

